### PR TITLE
Extend node APM stats with indices metrics

### PR DESCRIPTION
**Pull request by volodk85**
Extending node APM stats with indices metrics. 

Verified that new metrics are published in dev QA observability cluster

### CODE REVIEW

**Review by fcofdez — COMMENTED** [review]
Looks good, can we use the proper units?. Can we add the durations too? as well as indexing memory pressure?

File: server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java [hunk 1]
```diff
@@ -212,6 +212,70 @@ private void registerAsyncMetrics(MeterRegistry registry) {
...
+                "number",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getIndexFailedCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.deletion.docs.total",
+                "Total number of deleted documents",
+                "number",
```
[the remark this answers is not in the document]
Comment (volodk85) [reply server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java]: done

**Review by piergm — APPROVED** [review]
LGTM, thanks for adding these metrics. One small point we have to agree on is that in this PR https://github.com/elastic/elasticsearch/pull/103089  @rjernst cleaned up metrics names by removing `node.stats.` from it. If we decide to merge the linked PR we have to align this one too.

File: server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java [hunk 2]
```diff
@@ -212,6 +212,97 @@ private void registerAsyncMetrics(MeterRegistry registry) {
...
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.noop.total",
+                "Total number of noop shard operations",
+                "operations",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getNoopUpdateCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java:298]: Can we have all the indexing pressure stats exposed?

File: server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java [hunk 3]
```diff
@@ -212,6 +212,97 @@ private void registerAsyncMetrics(MeterRegistry registry) {
...
+                "Total number of deleted documents",
+                "documents",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getDeleteCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongGauge(
+                "es.node.stats.indices.deletion.docs.current",
+                "Current number of deleting documents",
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java:255]: `s/deleting/deleted/`

File: server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java (hunk 2 above)
Comment (volodk85) [reply server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java:298]: Added. Yes, skipping indexing replication for now.

File: server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java (hunk 3 above)
Comment (volodk85) [reply server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java:255]: is it grammar related comment? I've used *ing* to point on the fact that operation is ongoing

**Review by fcofdez — APPROVED** [review]
LGTM

**Pull request merged**